lib/Auth/TwoLeggedOAuth2.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace Mautic\Auth;
+
+use Mautic\Exception\IncorrectParametersReturnedException;
+use Mautic\Exception\RequiredParameterMissingException;
+
+/**
+ * OAuth Client modified from https://code.google.com/p/simple-php-oauth/.
+ */
+class TwoLeggedOAuth2 extends AbstractAuth
```

Comment:
Do you think we could make this class final and add real property, param and return types to avoid BC breaks when doing this later on?

Reply:
Agree with @nick-vanpraet

Reply:
Inheritance creates a multi-layered mess. A good example are Mautic controllers and models.

I suggest to google "composition over inheritance" to get good examples why composition is way better practice.

If the architecture is not allowing to change a class then Symfony allows you to [decorate the service](https://symfony.com/doc/current/service_container/service_decoration.html) which is not the point here but it is for Mautic.

Plus, a final class is easier to maintain for a library like this one as developers don't have to think about all the ways how users could have inherited the class and what it could break for them.

I'm not going to block this with these suggestions PR if it gets a second approval as I'm not using this library.